iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -288,11 +288,18 @@ private async Task Connect(IChannelHandlerContext context)
                     Debug.Assert(this.mqttTransportSettings.ClientCertificate != null);
                 }
 
+                string usernameString = $"{this.iotHubHostName}/{id}/?api-version={ClientApiVersionHelper.ApiVersionString}&DeviceClientType={Uri.EscapeDataString(this.productInfo.ToString())}";
+
+                if (!this.mqttTransportSettings.AuthenticationChain.IsNullOrWhiteSpace())
+                {
+                    usernameString += $"&auth-chain={Uri.EscapeDataString(this.mqttTransportSettings.AuthenticationChain)}";
```

Comment:
Why not use `AmqpIoTErrorAdapter.AuthChain` instead of `auth-chain` directly?

Reply:
AmqpIoTErrorAdapater.AuthChain has the : delimiter, which isn't used here for MQTT

Reply:
This is mqtt, my bad!